lib/AST/Attr.cpp

```diff
@@ -573,23 +575,27 @@ bool DeclAttribute::printImpl(ASTPrinter &Printer, const PrintOptions &Options,
     };
 
     // Print differentiation parameters, if any.
+    auto originalParamNames = map<SmallVector<Identifier, 8>>(
```

Comment:
Same here. It'd be more efficient if you just used `original->getParameters()->get(index)->getName().str()` in the loop.

Reply:
Done in https://github.com/apple/swift/pull/22000/commits/aec0072e4eb43e7c59c6b73d980030413deaceea.